docusaurus/docs/iOS/05-ui-cookbook/15-long-press-to-focus.mdx

```diff
@@ -0,0 +1,109 @@
+---
+title: Long Press to Focus
+description: Documentation on implementing a long press gesture to focus on a particular user in a StreamVideo call.
+---
+
+# Long Press to Focus
+
+The StreamVideo SDK allows for interactive and intuitive user engagement during calls. Implementing a long press gesture to focus your local participant's video feed can significantly enhance the user experience.
+
+## Overview
+
+- **Gesture Recognition**: Detect a long press on a participant's video feed.
+- **Focus Implementation**: Focus on the specified point on your video feed.
+- **User Experience**: Intuitive interaction for a more immersive call experience.
+
+## Implementing Long Press to Focus
+
+### Detecting the Long Press Gesture
+
+You can find an implementation that we are using in our Demo app that focus on longPress. To achieve that we are using the following ViewModifier
+```swift
+struct LongPressToFocusViewModifier: ViewModifier {
+
+    var availableFrame: CGRect
+
+    var handler: (CGPoint) -> Void
+
+    func body(content: Content) -> some View {
+        content
+            .gesture(
+                LongPressGesture(minimumDuration: 0.5)
+                    .sequenced(before: DragGesture(minimumDistance: 0, coordinateSpace: .local))
+                    .onEnded { value in
+                        switch value {
+                        case .second(true, let drag):
+                            if let location = drag?.location {
+                                handler(convertToPointOfInterest(location))
+                            }
+                        default:
+                            break
+                        }
+                    }
+            )
+    }
+
+    func convertToPointOfInterest(_ point: CGPoint) -> CGPoint {
+        CGPoint(
+            x: point.y / availableFrame.height,
+            y: 1.0 - point.x / availableFrame.width
+        )
+    }
+}
+```
+
+We can then define a View extension to allow us easily use the ViewModifier
```

Comment:
Seen this in few places, I think we should we put interpunction at the end, before showing the code samples.

Reply:
👍